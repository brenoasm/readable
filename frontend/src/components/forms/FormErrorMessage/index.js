import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const propTypes = {
  errors: PropTypes.arrayOf(PropTypes.string)
};

const defaultProps = {
  errors: []
};

const StyledFormErrorMessage = styled.div`
  span {
    color: #fe0303;
  }
`;

const FormErrorMessage = ({ errors }) => (
  <StyledFormErrorMessage>
    {errors && errors.map(error => (
      <span>{error}</span>
    ))}
  </StyledFormErrorMessage>
);

FormErrorMessage.propTypes = propTypes;
FormErrorMessage.defaultProps = defaultProps;

export default FormErrorMessage;
