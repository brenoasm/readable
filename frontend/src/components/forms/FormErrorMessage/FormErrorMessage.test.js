/* eslint-disable */

import React from 'react';

import FormErrorMessage from './';

describe('FormErrorMessage', () => {
  let wrapper;

  beforeEach(() => {
    const initialState = {
      errors: [
        'Erro ao preencher campo'
      ]
    };

    wrapper = shallow(<FormErrorMessage {...initialState} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have span length equals to initialState errors', () => {
    expect(wrapper.find('span')).toHaveLength(1);
  });
});
