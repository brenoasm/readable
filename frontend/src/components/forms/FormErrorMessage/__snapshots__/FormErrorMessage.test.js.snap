// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrorMessage matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormErrorMessage
    errors={
        Array [
            "Erro ao preencher campo",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <span>
          Erro ao preencher campo
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Erro ao preencher campo",
        },
        "ref": null,
        "rendered": "Erro ao preencher campo",
        "type": "span",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <span>
            Erro ao preencher campo
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Erro ao preencher campo",
          },
          "ref": null,
          "rendered": "Erro ao preencher campo",
          "type": "span",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
